core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4195,6 +4195,23 @@ public void doRestart(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
             rsp.sendRedirect2(".");
         }
     }
+    
+    /**
+     * REST client version of {@link #doRestart(StaplerRequest, StaplerResponse)}
+     * Perform a restart of Jenkins, if we can.
+     *
+     * This first replaces "app" to {@link HudsonIsRestarting}
+     *
+     * @since TODO
+     */
+    @RequirePOST
```

Comment:
Maybe restrict public API? Or do you want to use methods from outside the core?
```suggestion
    @RequirePOST
    @Restricted(NoExternalUse.class)
```